src/app/shared/components/category-select/category-select.component.ts

```diff
@@ -33,15 +43,52 @@ export class CategorySelectComponent implements OnInit {
   }
 
   ngOnInit(): void {
-    this.store.dispatch(new GetCategories());
     this.passCategoriesFormGroup.emit(this.CategoryFormGroup);
+    (this.workshop) && this.activateEditMode();
+  }
+
+  onFirstClick(): void {
+    if (this.categories.length < 2) {
+      this.store.dispatch(new GetCategories());
+      this.categories$.subscribe(cat => {
+        this.categories = cat;
+      });
+    }
   }
 
   onSelectCategory(category: Category): void {
     this.store.dispatch(new GetSubcategories(category.id));
+    this.subcategories$.subscribe(cat => {
+      this.subcategories = cat;
```

Comment:
use meaningful naming. do we initialize variable which is array of sub categories with single cat?